client/pkg/testutil/recorder.go

```diff
@@ -115,7 +116,10 @@ func (r *recorderStream) Chan() <-chan Action {
 
 func (r *recorderStream) Wait(n int) ([]Action, error) {
 	acts := make([]Action, n)
-	timeoutC := time.After(r.waitTimeout)
+	var timeoutC <-chan time.Time
+	if r.waitTimeout < math.MaxInt64 {
+		timeoutC = time.After(r.waitTimeout)
+	}
```

Comment:
Suggest to follow the pattern "`A Timeout of zero means no timeout.`"

FYI.
https://github.com/golang/go/blob/c2c4a32f9e57ac9f7102deeba8273bcd2b205d3c/src/net/http/client.go#L96

Reply:
@joshuazh-x  @ahrtr  I'd like to address this comment as we are seeing `TestPeriodic` tests flaking frequently in our CI. Please let me know if I can proceed or if there are any other plans to take it forward.

Reply:
If there is no feedback from @joshuazh-x next week, please feel free to create a new PR. thx

Reply:
@ahrtr  I am creating a new PR then with similar changes and review comments addressed. TY !

Reply:
OK, thanks

Reply:
I have cherry-picked the change and addressed review comments here, #18608 . Could you please take a look ?
thanks !